### PR TITLE
Integration Test cleanups

### PR DESCRIPTION
**Pull request by kuznetsovmoci**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -147,6 +147,16 @@ jobs:
   - script: |
       perl "%DDS_ROOT%\tests\cmake_integration\run_ci_tests.pl" --build-config=$(BuildConfiguration) --generator="Visual Studio 15 2017" --arch $(BuildPlatform)
     displayName: Compile and run CMake tests
+  - script: |
+      cd DevGuideExamples\DCPS\Messenger
+      perl appveyor_run_test.pl
```
Comment (mitza-oci) [inline azure-pipelines.yml]: Consider renaming this perl script.
Should a similar section be added to Linux and macOS configurations?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -314,3 +312,11 @@ jobs:
...
       $DDS_ROOT/tests/cmake_integration/run_ci_tests.pl
     displayName: Compile and run CMake tests
+  - script: |
+      cd DevGuideExamples/DCPS/Messenger
+      perl integration_run_test.pl
+    displayName: Run Examples DCPS Messenger
+  - script: |
+      cd tests/DCPS/Messenger
+      perl run_test.pl rtps
+    displayName: Run DCPS Messenger Test
```
Comment (iguessthislldo) [inline azure-pipelines.yml]: Github says the file is missing a final newline.

File: azure-pipelines.yml (hunk 1 above)
Comment (kuznetsovmoci) [reply azure-pipelines.yml]: I've added it for both, but the Test fails for MacOS (Example works fine).

File: dds/monitor/SPMonitorImpl.cpp [hunk 3]
```diff
@@ -46,8 +46,15 @@ SPMonitorImpl::report()
     report.host = this->hostname_.c_str();
     report.pid  = this->pid_;
     DDS::DomainParticipantFactory_var pf = TheParticipantFactory;
-    const DomainParticipantFactoryImpl::DPMap& participants =
-      dynamic_cast<DomainParticipantFactoryImpl*>(pf.in())->participants();
+
+    OpenDDS::DCPS::DomainParticipantFactoryImpl* pi = dynamic_cast<DomainParticipantFactoryImpl*>(pf.in());
```
Comment (mitza-oci) [inline dds/monitor/SPMonitorImpl.cpp]: Please make this a separate PR, it's not related to the other changes.

File: tests/DCPS/Messenger/run_test.pl [hunk 4]
```diff
@@ -29,6 +29,19 @@
 my $is_rtps_disc = 0;
 my $DCPSREPO;
 
+my $filename = "subscriber";
```
Comment (mitza-oci) [inline tests/DCPS/Messenger/run_test.pl]: Could this be changed in the common modules ($test->process or $test->start_process)?  It should apply to any test, and the common modules already know about file names extensions, etc.

File: .travis.yml [hunk 5]
```diff
@@ -89,6 +89,14 @@ script:
   - if [ "$SAFETY_PROFILE" != "1" ]; then
       $DDS_ROOT/tests/cmake_integration/run_ci_tests.pl;
     fi
+  - if [ "$NO_TESTS" == "1" ]; then
+      echo Building Messenger;
+      source setenv.sh;
```
Comment (mitza-oci) [inline .travis.yml]: I thought it already had the environment set

File: .travis.yml [hunk 6]
```diff
@@ -89,6 +89,14 @@ script:
...
+      cd tests/DCPS/Messenger;
+      mwc.pl -type gnuace;
```
Comment (mitza-oci) [inline .travis.yml]: You can combine this line with the previous one since `mwc.pl` takes a directory name as a command line argument.  It avoid the need to manage the `cd` state in different steps.

**Comment by mitza-oci** [comment]
I added a PR to vcpkg to fix the Xerces problem https://github.com/microsoft/vcpkg/pull/9702

File: .travis.yml (hunk 5 above)
Comment (kuznetsovmoci) [reply .travis.yml]: Without it it gives an error: "mwc.pl: command not found"

Comment (mitza-oci) [reply .travis.yml]: Right, it has some environment variables but not exactly the same as setenv.sh.  There's no need for it to be on the PATH when you know where to find it (`$ACE_ROOT/bin/mwc.pl`).

File: tests/DCPS/Messenger/run_test.pl (hunk 4 above)
Comment (kuznetsovmoci) [reply tests/DCPS/Messenger/run_test.pl]: I could be wrong, but it looks like common module functions like 'process' do not care about extension, as they can start the process without it. I can add this to the 'process' function:

  my $extension = ".exe";
  my $executable_ext = $executable + $extension;
  if (!(-e $executable) && !(-e $executable_ext)) {
    print STDERR "ERROR: executable \"$executable\" does not exist \n";
    $self->{status} = -1;
    return;
  }

Also, messenger script does not use 'process', or 'start_process', but 'PerlDDS::create_process'. I can either change that to use process / start_process or add similar check in PerlDDS::create_process

File: .travis.yml (hunk 6 above)
Comment (kuznetsovmoci) [reply .travis.yml]: This sequence was actually wrong. The correct one is:
      cd tests/DCPS/Messenger;
      mwc.pl -type gnuace .;
      make DDS_Messenger_Subscriber DDS_Messenger_Publisher;
      cd ../../..;

I am guessing I need to be in tests/DCPS/Messenger to correctly run nmake?

File: dds/monitor/SPMonitorImpl.cpp (hunk 3 above)
Comment (kuznetsovmoci) [reply dds/monitor/SPMonitorImpl.cpp]: Will do.

File: tests/DCPS/Messenger/run_test.pl (hunk 4 above)
Comment (mitza-oci) [reply tests/DCPS/Messenger/run_test.pl]: Some older scripts don't use "PerlDDS::TestFramework", you can migrate them to use it.

File: .travis.yml (hunk 6 above)
Comment (mitza-oci) [reply .travis.yml]: I wouldn't do it that way, but maybe that's just me.  I don't like managing the current directory state separately, it's something else that can go wrong:
```
$ACE_ROOT/bin/mwc.pl -type gnuace tests/DCPS/Messenger;
make -C tests/DCPS/Messenger
```

Comment (kuznetsovmoci) [reply .travis.yml]: I agree. This is done now.

File: .travis.yml (hunk 5 above)
Comment (kuznetsovmoci) [reply .travis.yml]: Fixed.

File: tests/DCPS/Messenger/run_test.pl (hunk 4 above)
Comment (kuznetsovmoci) [reply tests/DCPS/Messenger/run_test.pl]: Question: If I add the code above to 'process' sub, how would it stop the run_test script that called it, if 'process' fails? It sets $self->{status} = -1;, but I am not sure it does anything. I can start returning $self->{status}, and then check it in run_test.pl. Hope it won't break any existing scripts...

Comment (kuznetsovmoci) [reply tests/DCPS/Messenger/run_test.pl]: I have added the code above to 'process' sub.
